src/ripple/app/paths/PathRequest.cpp

```diff
@@ -380,7 +424,33 @@ int PathRequest::parseJson (Json::Value const& jvParams, bool complete)
                 uIss = *raSrcAccount;
             }
 
-            sciSourceCurrencies.insert ({uCur, uIss});
+            if (raSrcAccount && saSendMax)
```

Comment:
They way parsejson is called, the source account can never be null at this point.

I didn't want to disturb the code more than I needed to but I think I'll get rid of the unnecessary complete param as its the cause of some ugliness. The method is private and only called with it being true from a single place.